src/SignalR/server/StackExchangeRedis/src/RedisHubLifetimeManager.cs

```diff
@@ -784,6 +784,20 @@ private static string GenerateServerName()
         return $"{Environment.MachineName}_{Guid.NewGuid():N}";
     }
 
+    private static string GenerateInvocationId()
+    {
+        Span<byte> buffer = stackalloc byte[16];
+        var success = Guid.NewGuid().TryWriteBytes(buffer);
+        Debug.Assert(success);
+        // 16 * 4/3 = 21.333 which means base64 encoding will use 22 characters of actual data and 2 characters of padding ('=')
+        Span<char> base64 = stackalloc char[24];
+        success = Convert.TryToBase64Chars(buffer, base64, out var written);
+        Debug.Assert(success);
+        Debug.Assert(written == 24);
+        // Trim the two '=='
+        return new string(base64.Slice(0, base64.Length - 2));
```

Comment:
Fancy 😄 (probably less efficient though?)

```suggestion
        return new string(base64[..^2]);
```

Reply:
I dislike that syntax :/

Reply:
So much nicer